### ISSUE TITLE
error detected: Value '[]' at 'logEvents' failed to satisfy constraint: Member must have length greater than or equal to 1

### ISSUE THREAD

**Issue by chals-chu** [vectordotdev/vector#3795]
### Vector Version

```
vector 0.10.0 (g0f0311a x86_64-unknown-linux-gnu 2020-07-22)
...
```

### Vector Configuration File

```
[sources.linux]
  address = "0.0.0.0:514"
  mode = "udp"
  type = "syslog"

# Output data
[sinks.outcloudwatchlogs]
  encoding.codec = "json"
  type = "aws_cloudwatch_logs"
  inputs = ["linux"]
  compression = "none"
  create_missing_group = false
  create_missing_stream = true
  group_name = "gzone-kr-cw-logs"
  healthcheck = true
  region = "ap-northeast-2"
  stream_name = "gzone-kr-cw-logs-linux-%Y%m%d"
```

### Debug Output

```
Sep 10 13:30:01.410 TRACE sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}: vector::sinks::util::sink: adding event to batch.
Sep 10 13:30:01.410 TRACE source{name=linux type=syslog}: vector::internal_events::syslog: received line. byte_size=80
Sep 10 13:30:01.410 TRACE source{name=linux type=syslog}: vector::sources::syslog: processing one event. event=Log(LogEvent { fields: {"appname": Bytes(b"CROND"), "facility": Bytes(b"cron"), "host": Bytes(b"gm-logs-gw2"), "hostname": Bytes(b"gm-logs-gw2"), "message": Bytes(b"(root) CMD (/usr/lib64/sa/sa1 1 1)"), "procid": Integer(53905), "severity": Bytes(b"info"), "source_ip": Bytes(b"10.63.78.84:40128"), "source_type": Bytes(b"syslog"), "timestamp": Timestamp(2020-09-10T13:30:01Z)} })
Sep 10 13:30:01.410 TRACE sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}: vector::sinks::util::sink: adding event to batch.
Sep 10 13:30:02.401 TRACE sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}: vector::sinks::util::sink: batch linger expired.
Sep 10 13:30:02.401 TRACE sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}: vector::sinks::util::sink: submitting service request. in_flight_requests=1
Sep 10 13:30:02.401  INFO sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}:request{request_id=3}: vector::sinks::aws_cloudwatch_logs: Sending events. events=0
Sep 10 13:30:02.402 DEBUG http: vector::sinks::util::http: sending request. uri=https://logs.ap-northeast-2.amazonaws.com/ method=POST
Sep 10 13:30:02.501 DEBUG http: vector::sinks::util::http: response. status=200 version=HTTP/1.1
Sep 10 13:30:02.501 DEBUG sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}:request{request_id=3}: vector::sinks::aws_cloudwatch_logs::request: stream found stream=Some("gzone-kr-cw-logs-linux-20200910")
Sep 10 13:30:02.501  INFO sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}:request{request_id=3}: vector::sinks::aws_cloudwatch_logs::request: putting logs. token=None
Sep 10 13:30:02.502 DEBUG http: vector::sinks::util::http: sending request. uri=https://logs.ap-northeast-2.amazonaws.com/ method=POST
Sep 10 13:30:02.505 DEBUG http: vector::sinks::util::http: response. status=400 version=HTTP/1.1
Sep 10 13:30:02.505 ERROR sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}:request{request_id=3}: vector::sinks::util::retries: encountered non-retriable error. error=CloudwatchError::Put: 1 validation error detected: Value '[]' at 'logEvents' failed to satisfy constraint: Member must have length greater than or equal to 1
Sep 10 13:30:02.505 ERROR sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}:request{request_id=3}: vector::sinks::util::sink: Request failed. error=CloudwatchError::Put: 1 validation error detected: Value '[]' at 'logEvents' failed to satisfy constraint: Member must have length greater than or equal to 1
Sep 10 13:30:02.506 TRACE sink{name=outcloudwatchlogs type=aws_cloudwatch_logs}: vector::sinks::util::sink: acking events. acking_num=5
```

### Expected Behavior

The data received by syslog must be stored in the aws cloudwatch log stream.

### Actual Behavior

Received but not saved in the cloudwatch log.

### Example Data

```
centos 7.8 syslogs
(root) CMD (/usr/lib64/sa/sa1 1 1)
```

### Additional Context

```
If the source type is file, it is saved well in the cloudwatch log. However, if the source type is syslog, it cannot be saved in the cloudwatch log.
```

**Comment by chals-chu** [comment]
Install Vector From Source code git clone
vector 0.11.0 (g50a9c3a x86_64-unknown-linux-gnu 2020-09-10)
Not sent to syslog -> cloudwatch logs.

**Comment by juchiast** [comment]
Thanks for reporting! 

Duplication of #3707

**Comment by jamtur01** [comment]
Closed as up of #3707.
